galahq/gala
# chore: rails 6.1, postgres 16, redis 7 tls updates

The Postgres and Redis instances require maintenance.

> Heroku Postgres 13 is being deprecated on October 29, 2024 - [heroku.com/changelog-items/2778](https://devcenter.heroku.com/changelog-items/2778).
> Heroku Redis 6 is being deprecated on Dec 2, 2024 and requires TLS - [heroku.com/changelog-items/2992](https://devcenter.heroku.com/changelog-items/2992)

This PR upgrades Postgres and removes dependency on Redis and Sidekiq.

Deployed changes:

- updated docker-compose.yml file to postgres 16 and redis 7 to align with heroku instances
- added a worker to Profile.dev to make docker images/containers smaller
- upgrade to rails 6.1 (prepping for heroku stack-22 upgrade)
- removed the memcached dependency in favor of OSS redis as it can handle caching, pub/sub for websockets, and an enqueuing backend for sidekiq. This also will make updating gems, ruby and rails easier for the upcoming
- updated the Dockerfile using techniques from 37signals to make the local docker environment faster and is production ready should we need to switch to deploying a docker images instead of heroku buildpacks
- includes the migration for Postgres 16 upgrade that was already ran in production
- includes two other migrations created by rails for the rails 6.1 update related to active storage
- added the rake test:unit to run only the unit tests

To test:

- prune docker, (docker compose down, docker system prune, docker volume prune, docker volume rm gala_db_data)
- start app with docker compose up --build
- ensure app builds and can be accessed at http://localhost:3000/, you should see seed data
- run docker compose run web rake indices:refresh inside container to ensure it works
- trying searching for a case and ensure you are able to see results
- inspect docker console, ensure you see no postgres or redis errors while performing exploratory testing in the app
- ensure unit tests still pass, docker compose run web rake test:unit
- ensure copying a case works and creating a translated case works.
